content/PHPDataObjects/DataObjects.q.md

```diff
@@ -0,0 +1,1241 @@
+
+/// type=MS, answer=[1,2,3,4,5]
+
+What are the column names in the `students` table?
+
+- `gender`
+
+- `birth_date`
+
+- `last_name`
+
+- `first_name`
+
+- `student_id`
+
+
+/// type=SS, answer=[5]
+
+Which statement correctly describes the code on lines 12, 13, and 14 of `student.php`?
+
+- It creates a new `Exception` class.
+
+- It updates the `Exception` message thrown.
+
+- It adds an exception in the statement `echo $e->getMessage()`.
+
+- It removes an exception using the statement `echo $e->getMessage();`.
+
+- It executes the `echo` statement if an exception occurs within the `try` block.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[connection.php,student.php], init=[commands/DataObjects/InsertJohnSmith.sql]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. ";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    try {
+        $stmt = $conn->query("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')");
+        if (!$stmt) {
+            throw new Exception("Unable to insert values into the table.");
+        }
+        echo "Successfully inserted the values into the table.";
+    } catch (Exception $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`
+
+- It prints `Successfully connected to the database.` and `Unable to insert values into the table.`.
+
+- It prints `Successfully connected to the database.` and `Successfully inserted the values into the table.`.
+
+
+/// type=MS, answer=[4,5]
+
+Which statements correctly describe the code on line 9 of `connection.php`?
+
+- It evaluates the `$conn` object.
+
+- It sets the `PDO` class to the `PDO_PGQSL DSN` statement
+
+- It assigns the `PDO_PGQSL DSN` statement to the `PDO` class.
+
+- It creates the `$conn` object as an instance of the `PDO` class.
+
+- It passes the `PDO_PGQSL DSN` statement as an argument of the `PDO` class.
+
+
+/// type=MS, answer=[1,2,3,5]
+
+What values are inserted by the statement `INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')")` on line 5?
+
+- `Male`
+
+- `John`
+
+- `Smith`
+
+- `first_name`
+
+- `1999-02-10`
+
+
+/// type=SS, answer=[5]
+
+In `student.php`, what does the query `INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')")` do on line 5?
```

Comment:
Typo:

Please remove __`")`__ from the `INSERT INTO` statement here.... Thanks.